src/Filter/ORM/Equals.php

```diff
@@ -26,19 +26,22 @@ public function filter($queryBuilder, $metadata, $option)
             $option['alias'] = 'row';
         }
 
+        [$typeCastMetaData, $typeCastFieldName] = $this->getTypeCastParams($queryBuilder, $metadata, $option['field'], $option['alias']);
+        $fieldType = $typeCastMetaData->getTypeOfField($typeCastFieldName);
+
         $format = null;
         if (isset($option['format'])) {
             $format = $option['format'];
         }
 
-        $value = $this->typeCastField($metadata, $option['field'], $option['value'], $format);
+        $value = $this->typeCastField($typeCastMetaData, $typeCastFieldName, $option['value'], $format);
```

Comment:
Metadata is a single word in Doctrine https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/metadata-drivers.html